### PR TITLE
fix: api parsing

### PR DESCRIPTION
**Pull request by umerm-work**
### Checklist

- [x] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md).
- [ ] My change requires a documentation update, and I have done it.
- [ ] My change requires updating the Open API specification and/or changing its version, and I've done it.
- [x] I have added tests to cover my changes.
- [ ] I have filled out the description and linked the related issues.

### Description
This PR fixes parsing issues in (3214)

### Screenshots (if appropriate):

- - -
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/3311)

### CODE REVIEW

File: pkg/api/validation_test.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+			name: "address",
+			input: map[string]string{
+				"address": "bff2c89e85e78c38bd89fca1acc996afb876c21bf5a8482ad798ce15f1c223fa",
+			},
+			output:  &parseAddressTest{Address: []byte("bff2c89e85e78c38bd89fca1acc996afb876c21bf5a8482ad798ce15f1c223fa")},
+			wantErr: nil,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
```
Comment (notanatol) [inline pkg/api/validation_test.go:82]: we might as well run this in parallel, with the new upcoming linter

File: pkg/api/validation.go [hunk 2]
```diff
@@ -54,51 +66,43 @@ func (s *Service) parseAndValidate(input *http.Request, output interface{}) (err
...
+		//		return GetErrorMessage(propertyName, errMessage)
+		//	}
+		//	if len(body) > 0 {
+		//		err = json.Unmarshal(body, &output)
+		//		if err != nil {
+		//			return GetErrorMessage(propertyName, errMessage)
+		//		}
+		//
+		//	}
+		//}
```
Comment (notanatol) [inline pkg/api/validation.go:81]: let's try not to leave commented out code

**Comment by notanatol** [comment]
when this is merged make sure the title says something like `fix: api parsing and validation`

File: pkg/api/validation_test.go (hunk 1 above)
Comment (umerm-work) [reply pkg/api/validation_test.go:82]: @notanatol could you please share a sample or reference of the linters?

File: pkg/api/validation_test.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+		{
+			name: "address",
+			input: map[string]string{
+				"address": "bff2c89e85e78c38bd89fca1acc996afb876c21bf5a8482ad798ce15f1c223fa",
+			},
+			output:  &parseAddressTest{Address: []byte("bff2c89e85e78c38bd89fca1acc996afb876c21bf5a8482ad798ce15f1c223fa")},
+			wantErr: nil,
+		},
+	}
+	for _, tt := range tests {
```
Comment (notanatol) [inline pkg/api/validation_test.go:82]: dont forget to add this line:
```
tt := tt
```
otherwise the `t.Run` will pick up only the last test case

**Review by notanatol — APPROVED** [review]

File: pkg/api/validation_test.go (hunk 1 above)
Comment (notanatol) [reply pkg/api/validation_test.go:145]: the linters are being added one by one by vlado, here's the reference on how to do it: https://github.com/ethersphere/bee/blob/master/CODINGSTYLE.md#parallel-test-execution

File: pkg/api/validation_test.go [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+				"address": "bff2c89e85e78c38bd89fca1acc996afb876c21bf5a8482ad798ce15f1c223fa",
+			},
+			want:    &parseAddressTest{Address: decodeString},
+			wantErr: nil,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt := tt
+			indirect := reflect.Indirect(reflect.ValueOf(tt.want))
```
Comment (notanatol) [inline pkg/api/validation_test.go:147]:
```suggestion
	for _, tt := range tests {
		tt := tt
		t.Run(tt.name, func(t *testing.T) {
			t.Parallel()
			indirect := reflect.Indirect(reflect.ValueOf(tt.want))
```

**Review by notanatol — APPROVED** [review]

**Comment by mrekucci** [comment]
Closed in favor of #3344

**Pull request closed without merging**